### PR TITLE
Add new assertion to have better info about test failure

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/17001

Modifications:
- added new assertion to have better info about test failure
- refactored checkPositive utility method
- refactored cyclic buffer size check

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1025/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1025/Hazelcast-pr-builder-1025.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientMapEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR] Errors: 
[ERROR]   CyclicBufferTest.test_size:191 ? IllegalArgument sequence but 0
[ERROR]   ReadOnlyIteratorTest.setUp:45 ? IllegalArgument sequence but 0
[ERROR]   ReadOnlyIteratorTest.setUp:45 ? IllegalArgument sequence but 0
[ERROR]   ReadOnlyIteratorTest.setUp:45 ? IllegalArgument sequence but 0
[INFO] 
[ERROR] Tests run: 27745, Failures: 1, Errors: 4, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java [hunk 1]
```diff
@@ -33,37 +33,38 @@
  */
 public class DefaultCyclicBuffer<E extends Sequenced> implements CyclicBuffer<E> {
 
-    private static final long UNAVAILABLE = -1L;
+    private static final long UNSET = -1L;
 
     private int capacity;
     private E[] buffer;
     private AtomicLong headSequence;
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java:40]: This smells like a design problem. The class is lock free and now get some locks.

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java [hunk 2]
```diff
@@ -33,37 +33,38 @@
  */
 public class DefaultCyclicBuffer<E extends Sequenced> implements CyclicBuffer<E> {
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java:34]: Can you explain from which threads this buffer is accessed, how and when? Why the need for such coarse synchronization?

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java:40]: put a new assertion to have much info about failure and removed locks

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java (hunk 2 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java:34]: put a new assertion to have much info about failure and removed synchronization. This class is accessed by partition threads.

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java [hunk 3]
```diff
@@ -137,16 +138,11 @@ public int size() {
...
-        if (tail == UNAVAILABLE) {
+        if (tail == UNSET) {
             return 0;
         }
 
-        int avail = (int) (tail - head + 1);
-        if (avail <= capacity) {
-            return avail;
-        } else {
-            return capacity;
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java:148]: How was it even possible in the previous code that the available was higher than the capacity?

**Pull request merged**